contrib/scripts/unloaddb.sh

```diff
@@ -0,0 +1,402 @@
+                show_usage
+                exit
+        fi
+
+        database=$(echo ${@:$nelem:1})
+}
+
+function get_table_size ()
+{
+        local num_rows=0
+        local table_name=$1
+        local row_size=0
+        local table_size=0
+
+        num_rows=$(csql $user $pass -l -c "show heap capacity of $table_name" $db | grep Num_recs | awk '{print $3}')
+        row_size=$(get_schema_size $table_name)
+        let "table_size = num_rows * row_size"
+
+        echo $table_size
+}
+
+function get_schema_size ()
+{
+        local table_name=$1
+        local size=0
+        local query=" select CAST(SUM(CASE \
+         WHEN "data_type" = 'BIGINT' THEN 8.0 \
+         WHEN "data_type" = 'INTEGER' THEN 4.0 \
+         WHEN "data_type" = 'SMALLINT' THEN 2.0 \
+         WHEN "data_type" = 'FLOAT' THEN 4.0 \
+         WHEN "data_type" = 'DOUBLE' THEN 8.0 \
+         WHEN "data_type" = 'MONETARY' THEN 12.0 \
+         WHEN "data_type" = 'STRING' THEN prec \
+         WHEN "data_type" = 'VARCHAR' THEN prec \
+         WHEN "data_type" = 'NVARCHAR' THEN prec \
+         WHEN "data_type" = 'CHAR' THEN prec \
+         WHEN "data_type" = 'NCHAR' THEN prec \
+         WHEN "data_type" = 'TIMESTAMP' THEN 8.0 \
+         WHEN "data_type" = 'DATE' THEN 4.0 \
+         WHEN "data_type" = 'TIME' THEN 4.0 \
+         WHEN "data_type" = 'DATETIME' THEN 4.0 \
+         WHEN "data_type" = 'BIT' THEN FLOOR(prec / 8.0) \
+         WHEN "data_type" = 'BIT VARYING' THEN FLOOR(prec / 8.0) \
+         ELSE 0 \
+     END) as BIGINT)  AS [size] \
+ from db_attribute where class_name = '$table_name';"
+
+        if [ $# -eq 0 ];then
+                echo "0"
+                return
+        fi
+
+        size=$(csql $user $pass -l -c "$query" $db | grep "^<0000" | awk '{print $3}')
+        echo $size
+}
+
+function find_slot ()
+{
+        local selected=0
+        local i
+        local size=${slot_size[0]}
+
+        for ((i = 0; i < $num_proc; i++))
+        do
+                if [ ${slot_size[i]} -lt $size ];then
+                        size=${slot_size[i]}
+                        selected=$i
+                fi
+        done
+
+         echo $selected
+}
+
+function do_unloaddb ()
+{
+        local slot_num=$1
+        local i
+        local file="Tables_unloaded_$database.$slot_num"
+        local num_tables_in_slot=0
+        local msg="Success"
+        local pid
+
+        for ((i = 0; i < $num_tables; i++))
+        do
+                if [ ${slot[i]} -eq $slot_num ];then
+                        let "num_tables_in_slot++"
+                        echo "${table_selected[i]}" >> $file
+                fi
+        done
+
+        if [ $verbose = "yes" ];then
+                echo "Proc $slot_num: num tables: $num_tables_in_slot, ${slot_size[$slot_num]} bytes"
+        fi
+
+        echo "process $slot_num: starting" > unloaddb_$slot_num.status
+
+	cubrid unloaddb $user $pass --input-class-only --input-class-file $file $database &
```

Comment:
How about using the -d or --data-only option here?
And, I think that it would be better not to unload schema files at this script.
If it is necessary to unload schema file, it is better to unload it all at once rather than dividing it.

Reply:
In that case, neither index nor trigger could be unloaded as well, what do you think?

Reply:
I think that it should be handled with trigger and index files in the same way as schema file.
If we decide that it don't be unloaded the schema file at this shell script, trigger and index won't need to unloaded too.